### PR TITLE
Fix: restatectl dump cluster renders status for all live nodes

### PR DESCRIPTION
**Pull request by pcholakov**
This change fixes a trivial issue in the dump cluster sub-command where it would only print the partition processor status for one node rather than all nodes scheduler to run a given partition.

It also renames the tool from its original `dump cluster-status` subcommand to `partitions status`, which reflects much better what it's doing.

The server-side change is purely a cosmetic refactoring replacing an early if/continue with a two-arm match.

Example from a local three-worker / 4 partitions cluster:

```
> restatectl partitions status --cluster-controller $RESTATE_CLUSTER_CONTROLLER_ADDRESS
Alive partition processors (nodes config v7, partition table v1)
 P-ID  NODE  RUN-MODE          REPLAY  APPLIED-LSN  PERSISTED-LSN  OBSERVED-LEADER  #-SKIPS  LAST-SEEN
 0     N2:1  Leader            Active  2            ??             N2:1 - e2        0        1 second and 288 ms ago
 0     N4:1  Follower          Active  2            ??             N2:1 - e2        0        1 second and 381 ms ago
 1     N2:1  Leader            Active  1            ??             N2:1 - e1        0        1 second and 181 ms ago
 1     N4:1  Follower          Active  0            ??             ?? - ??          0        995 ms ago
 2     N2:1  Leader            Active  1            ??             N2:1 - e1        0        890 ms ago
 2     N4:1  Follower          Active  0            ??             ?? - ??          0        1 second and 425 ms ago
 3     N2:1  Follower          Active  1            ??             N3:1 - e1        0        1 second and 197 ms ago
 3     N4:1  Follower->Leader  Active  1            ??             N3:1 - e1        0        979 ms ago

☠️ Dead nodes
 NODE  LAST-SEEN
 N3    7 minutes, 5 seconds and 222 ms ago```
```

I'm keeping the `Dump` command around as it will be useful very shortly, I'll add the ability to do a raw dump of nodes / partition table / logs config there in addition to the friendlier tools we now have.

### CODE REVIEW

**Comment by AhmedSoliman** [comment]
Perhaps this can be `partitions status` instead of dump cluster-state then?

**Review by AhmedSoliman — CHANGES_REQUESTED** [review]
Marking as as "request changed" to get it back to your queue.

File: tools/restatectl/src/commands/dump/mod.rs [hunk 1]
```diff
@@ -8,12 +8,7 @@
...
-
 use cling::prelude::*;
 
-#[derive(Run, Subcommand, Clone)]
-pub enum Dump {
-    /// Dump the latest cluster state
-    ClusterState(cluster_state::ClusterStateOpts),
-}
+#[derive(Subcommand, Clone)]
+pub enum Dump {}
```
Comment (pcholakov) [inline tools/restatectl/src/commands/dump/mod.rs:14]: I'll come back to this shortly.

**Comment by pcholakov** [comment]
Updated as suggested!

**Review by AhmedSoliman — APPROVED** [review]
🚢

**Pull request merged**